lib/mocks/jobs.coffee

```diff
@@ -12,15 +12,25 @@ class MockJob extends EventEmitter
   intervalEmit: (event, timeout, data) ->
     interval = Math.floor timeout / (data.length + 1)
     @delayEmit event, (i + 1) * interval, el for el, i in data
+  # Takes a list of events and returns a new list of events sorted in increasing order, with each
+  # timeout being the offset from the previous event, such that the total offset for the event is
+  # the same.
+  @orderEvents: (events) ->
+    return events unless events.length > 1
+    events = _(events).sortBy 'timeout'
+    # Pair each event with the previous event, except for the first which gets paired with 0
+    pairs = _.zip events, [timeout: 0].concat _.first(events, events.length - 1)
+    _.map pairs, ([curr, prev]) ->
+      _.extend {}, curr, timeout: curr.timeout - prev.timeout
   start: =>
-    events =
-      if @events.length < 2
-        @events
-      else
-        events = _.sortBy @events, (e) -> e.timeout
-        event_pairs = _.zip events, _.rest(events).concat [_.last events]
-        _.map event_pairs, ([event, next_event]) ->
-          _.extend event, timeout: next_event.timeout - event.timeout
+    # We can't just set timeouts for all of the events because it may not result in a predictable
+    # order. For instance, consider:
+    # a 'complete' event is registered after 100ms
+    # a 'created' event is registered after 10ms
+    # If something then blocks the event loop for 110ms, both will fire on the next tick, in the
+    # order that they were registered, so we'd see the 'complete' event before seeing the 'create'
+    # event.
+    events = @constructor.orderEvents @events
```

Comment:
It could be `MockJob.orderEvents`. It could also be a method that's exported by the function, but that wasn't consistent with the fact that the module currently only exports various implementations of `MockJob`